paasta_tools/contrib/reset_kubernetes_cluster.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/python3.7
+import argparse
+from os import environ
+from random import choice
+from subprocess import run
+from textwrap import indent
+from typing import List
+from urllib.parse import urlparse
+
+import yaml
+
+ALLOWED_CLUSTERS = ["kubestage"]
+
+
+def inv_search(*filters: str) -> List[str]:
+    """Search inventory service and return matching hosts"""
+    _filters = [f"-F {item}" for item in filters]
+    return run(["inv", "find"] + _filters, capture_output=True).stdout.decode().split()
+
+
+class KubernetesResetter:
+    """Reset the cluster."""
+
+    dry: bool
+
+    master_node_names: List[str]
+    other_node_names: List[str]
+
+    etcd_prefix: str
+    etcd_host: str
+
+    cluster_name: str
+    cluster_ecosystem: str
+
+    def __init__(self, dry=True) -> None:
+        self.dry = dry
+        self.master_node_names = []
+        self.other_node_names = []
+        with open("/nail/etc/ecosystem", "r+") as _eco:
+            self.cluster_ecosystem = _eco.read().strip()
```

Comment:
Why are we reading the ecosystem from the file instead of having it read from stdin? That means that the file will only be executable on nodes in the same ecosystem. Why?

Reply:
The script needs access to the kubeadm config, so it has to be run on a kubernetes master, hence the master's ecosystem is assumed.